managed/services/telemetry/transform_test.go

```diff
@@ -107,48 +107,171 @@ func Test_transformToJSON(t *testing.T) {
+func TestTransformStripValues(t *testing.T) {
+	type args struct {
+		config  *Config
+		metrics []*pmmv1.ServerMetric_Metric
+	}
+
+	noMetrics := []*pmmv1.ServerMetric_Metric{}
+
+	tests := []struct {
+		name    string
+		args    args
+		want    []*pmmv1.ServerMetric_Metric
+		wantErr assert.ErrorAssertionFunc
+	}{
+		{
+			name: "nil metrics",
+			args: args{
+				config:  configEnvVars(),
+				metrics: nil,
+			},
+			want:    nil,
+			wantErr: assert.NoError,
+		},
+		{
+			name: "empty metrics",
+			args: args{
+				config:  configEnvVars(),
+				metrics: noMetrics,
+			},
+			want:    noMetrics,
+			wantErr: assert.NoError,
+		},
+		{
+			name: "no Transform in config",
+			args: args{
+				config:  configEnvVars().noTransform(),
+				metrics: noMetrics,
+			},
+			want:    noMetrics,
+			wantErr: assert.NoError,
+		},
+		{
+			name: "no Metrics config",
+			args: args{
+				config:  configEnvVars().noFirstMetricConfig(),
+				metrics: noMetrics,
+			},
+			want:    noMetrics,
+			wantErr: assert.NoError,
+		},
+		{
+			name: "no Metric Name config",
+			args: args{
+				config: configEnvVars().noFirstMetricNameConfig(),
+				metrics: []*pmmv1.ServerMetric_Metric{
+					{Key: "metric-a", Value: "v1"},
+					{Key: "metric-b", Value: "v2"},
+				},
+			},
+			want: []*pmmv1.ServerMetric_Metric{
+				{Key: "metric-a", Value: "1"},
+				{Key: "metric-b", Value: "1"},
+			},
+			wantErr: assert.NoError,
+		},
+		{
+			name: "invalid data source",
+			args: args{
+				config: configEnvVars().changeDataSource(dsPMMDBSelect),
+				metrics: []*pmmv1.ServerMetric_Metric{
+					{Key: "metric-a", Value: "v1"},
+					{Key: "metric-b", Value: "v2"},
+				},
+			},
+			want:    nil,
+			wantErr: assert.Error,
+		},
+		{
+			name: "happy path",
+			args: args{
+				config: configEnvVars(),
+				metrics: []*pmmv1.ServerMetric_Metric{
+					{Key: "metric-a", Value: "v1"},
+					{Key: "metric-b", Value: "v2"},
+				},
+			},
+			want: []*pmmv1.ServerMetric_Metric{
+				{Key: "metric-a", Value: "1"},
+				{Key: "metric-b", Value: "1"},
+			},
+			wantErr: assert.NoError,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := transformStripValues(tt.args.config, tt.args.metrics)
+			if !tt.wantErr(t, err) {
+				t.Logf("config: %v", tt.args.config)
+				return
+			}
+			assert.Equalf(t, tt.want, got, "transformStripValues(%v, %v)", tt.args.config, tt.args.metrics)
+		})
+	}
+}
+
+func configJSON() *Config {
 	return &Config{
 		Transform: &ConfigTransform{
 			Metric: "metric",
-			Type:   JSONTransformType,
+			Type:   JSONTransform,
 		},
 		Data: []ConfigData{
 			{MetricName: "my-metric", Label: "label"},
 		},
 	}
 }
 
+func configEnvVars() *Config {
+	return &Config{
+		Source: "ENV_VARS",
+		Transform: &ConfigTransform{
+			Type: StripValuesTransform,
+		},
+		Data: []ConfigData{
+			{MetricName: "ENABLE_A", Column: "column-a"},
+			{MetricName: "DISABLE_B", Column: "column-b"},
+		},
```

Comment:
this data field confuses in context of this test, seems like it's not used and can be removed

Reply:
Agree... I had to yank one useless test that was relying on the `data` field )